### ISSUE TITLE
Subnetwork Module Feature Request

### ISSUE THREAD

**Issue by jab0707** [SCIInstitute/SCIRun#2160]
Sometimes there are blocks of modules that it would be useful to run as a block, making organization easier as well as allowing for more complicated network design.

It would hopefully look something like this in the main view:
![SubNet-1](https://user-images.githubusercontent.com/38257234/88201246-76020200-cc04-11ea-80d7-04fd3d5ce153.png)
Here it would have dynamic field, matrix, and string inputs. (Dynamic outputs would be great too but even a fixed number would be fine for now)

Upon opening the UI for the subnetwork it would pull up a window with another module editor view like below:
![SubNet0](https://user-images.githubusercontent.com/38257234/88201247-76020200-cc04-11ea-90f6-96e380d385ff.png)
Note specifically that at the top of the subnetwork there are the input ports and at the bottom there are the output ports. (This could also be accomplished by a module in the subnetwork called "Pull subnetwork inputs" and "push subnetwork outputs")

When filled out with some modules the subnetwork view should look like this:
![PythonSubNet](https://user-images.githubusercontent.com/38257234/88201237-74d0d500-cc04-11ea-8b87-ee6e6f6a0eeb.png)

Finally it would be particularly useful if the subnetwork could be executed and its inputs pulled via a python command as shown below with one example just running it and another also pulling results.
![SubNet1](https://user-images.githubusercontent.com/38257234/88201248-769a9880-cc04-11ea-8078-11b08d95a7d9.png)

**Comment by jessdtate** [comment]
have you used the subnet implementation in SR5 before?  It may be currently off because of build problems.  

existing issue: #1652

**Comment by jab0707** [comment]
I have not tried it. Is it just a cmake option? I can try building with it.

**Comment by dcwhite** [comment]
Yes, these images look eerily like the existing disabled implementation. It's turned off in the code since it wasn't working well. I can turn it on and you can find all 532 bugs with it very quickly.

**Comment by we85** [comment]
Are there any updates on this matter?

**Comment by jab0707** [comment]
I will mention that it would still be useful

**Comment by dcwhite** [comment]
#2177 is going to be worked on soon, which will provide most of the benefits of subnets while being easier to implement. Stay tuned.

**Comment by dcwhite** [comment]
**Analysis (2026-06-13)**

Subnetwork module feature request. The existing subnet implementation is disabled (CMake flag, not working well). Per comments, #2177 (composite module builder) is the path forward and is nearly ready. This issue can be closed in favor of #2177 once composite modules are stable.
